envi/config.py

```diff
@@ -35,6 +35,30 @@ def getusername():
     type(None):(int,str,bool,long,unicode),
 }
 
+CONFIG_PATH = 0
+CONFIG_ENTRY = 1
+
+class ConfigNoAssignment(Exception):
+    def __init__(self, optstr):
+        Exception.__init__(self)
+        self.optstr = optstr
+    def __str__(self):
+        return "No value given in option %s" % self.optstr
+
+class ConfigInvalidName(Exception):
+    def __init__(self, optpath):
+        Exception.__init__(self)
+        self.optpath = optpath
+    def __str__(self):
+        return 'Invalid Config Name: %s' % self.optpath
+
+class ConfigInvalidOption(Exception):
+    def __init__(self, optname):
+        Exception.__init__(self)
+        self.optname = optname
+    def __str__(self):
+        return 'Invalid Config Option: %s' % self.optname
+
```

Comment:
These look great. Once a couple nits are fixed, we should be able to mainline it easily.